doc/release/1.14.5-notes.rst

```diff
@@ -0,0 +1,30 @@
+==========================
+NumPy 1.14.5 Release Notes
+==========================
```

Comment:
A little surprised to see forward-porting release notes from one branch into the next like this.

I think this would make sense as a logically separate commit.

Reply:
I thought about it, but the 1.14.5 notes were so small I figured to slip them in here. The oddity was the timing, I probably should have waited another day for the branch.